orttraining/orttraining/models/runner/training_runner.cc

```diff
@@ -388,14 +397,92 @@ Status TrainingRunner::TrainingLoop(IDataLoader& training_data_loader, IDataLoad
+  }
+
   std::cout << "Round: " << round_ << "\n"
             << "Batch size: " << params_.batch_size << "\n"
-            << "Number of Batches: " << (step_ - step_start) << "\n"
+            << "Number of Batches: " << number_of_batches << "\n"
             << "Gradient Accumulation Steps: " << gradient_accumulation_step_count << "\n"
-            << "Weight Update Steps: " << (weight_update_step_count_ - weight_update_step_count_start) << "\n"
+            << "Weight Update Steps: " << weight_update_steps << "\n"
             << "Total Running Time: " << total_time << " Seconds \n"
-            << "Average Running Time Per Batch: " << total_time / (step_ - step_start) * 1000 << " ms\n"
-            << "Throughput: " << params_.batch_size * (step_ - step_start) / total_time << " Examples / Second\n";
+            << "Average Running Time Per Batch: " << avg_time_per_batch << " ms\n"
+            << "Throughput: " << throughput << " Examples / Second\n";
+
+  return Status::OK();
+}
+
+Status TrainingRunner::SavePerfMetrics(const size_t number_of_batches, const size_t gradient_accumulation_steps,
+                                       const size_t weight_update_steps, const double total_time,
+                                       const double avg_time_per_batch, const double throughput) {
+  // popualte metrics for reporting
+  json j;
+  j["Model"] = params_.model_type;  
+  auto it = params_.perf_properties.find("sequence");
+  std::string seq_len;
+  if (it != params_.perf_properties.end()) {
+    j["SeqLen"] = it->second;
+    seq_len = std::to_string(it->second);
+  }
+  it = params_.perf_properties.find("dynamic_prediction_count");
+  if (it != params_.perf_properties.end())
+    j["PredictionsPerSeq"] = it->second;
+  j["Round"] = round_;
+  j["BatchSize"] = params_.batch_size;
+  j["NumOfBatches"] = number_of_batches;
+  j["GradAccSteps"] = gradient_accumulation_steps;
+  j["WeightUpdateSteps"] = weight_update_steps;
+  j["TotalTime"] = total_time;
+  j["AvgTimePerBatch"] = avg_time_per_batch;
+  j["Throughput"] = throughput;
+  j["MixedPrecision"] = params_.use_mixed_precision ? "fp16" : "fp32";
+  j["Optimizer"] = params_.training_optimizer_name;
+
+  // TODO - add memory/cpu
+  //j["Memory"] = ;
+  //j["AvgCPU"] = ;
+
+  //
+  // we will get date/time and commitId in post-run pipeline
+  //          
+
+  // populate other basic params for bookkeeping - add more as needed
+  json p;
+  p["LearningRate"] = params_.lr_params.initial_lr;
+  p["WarmupRatio"] = params_.lr_params.warmup_ratio;
+  p["WarmupMode"] = params_.lr_params.warmup_mode;
+  p["TrainSteps"] = params_.num_train_steps;
+  p["ModelPath"] = ToMBString(params_.model_path.c_str());
+  p["TrainDataDir"] = ToMBString(params_.train_data_dir.c_str());
+  p["TestDataDir"] = ToMBString(params_.test_data_dir.c_str());
+
+  j["RunConfig"] = p.dump(); // serialize the params as json string
+
+  std::string jsonString = j.dump(); 
```

Comment:
naming convention `json_string`. also the names j, p, and jsonString could be more descriptive :)